diff -aupr a/nvidia390.spec b/nvidia390.spec
--- a/nvidia390.spec	2020-03-10 15:51:03.000000000 +0300
+++ b/nvidia390.spec	2020-04-11 17:02:17.419991326 +0300
@@ -89,7 +89,7 @@
 Summary:	NVIDIA proprietary X.org driver and libraries
 Name:		%{name}
 Version:	%{version}
-Release:	1
+Release:	1magos
 
 Source0:	http://download.nvidia.com/XFree86/Linux-x86/%{version}/%{pkgname32}.run
 Source1:	http://download.nvidia.com/XFree86/Linux-x86_64/%{version}/%{pkgname64}.run
@@ -794,10 +794,10 @@ done
 
 # Create the convenience symlinks. As the packages with nvidia are mutually
 # exclusive now, there is no need to use alternatives for this.
-ln -s %{nvidia_libdir} %{buildroot}%{_libdir}/nvidia
-%ifarch %{biarches}
-	ln -s %{nvidia_libdir32} %{buildroot}%{_prefix}/lib/nvidia
-%endif
+#ln -s %{nvidia_libdir} %{buildroot}%{_libdir}/nvidia
+#%ifarch %{biarches}
+#	ln -s %{nvidia_libdir32} %{buildroot}%{_prefix}/lib/nvidia
+#%endif
 
 # confirm SONAME; if something else than libvdpau_nvidia.so or libvdpau_nvidia.so.1, adapt .spec as needed:
 [ "$(objdump -p %{buildroot}%{nvidia_libdir}/vdpau/libvdpau_nvidia.so.%{version} | grep SONAME | gawk '{ print $2 }')" = "libvdpau_nvidia.so.1" ]
@@ -931,6 +931,12 @@ mkdir -p %{_libdir}/vdpau
 	--slave %{_bindir}/nvidia-modprobe nvidia-modprobe %{nvidia_bindir}/nvidia-modprobe \
 	--slave %{_bindir}/nvidia-persistenced nvidia-persistenced %{nvidia_bindir}/nvidia-persistenced \
 	--slave %{_sysconfdir}/X11/xinit.d/nvidia-settings.xinit nvidia-settings.xinit %{_sysconfdir}/%{drivername}/nvidia-settings.xinit \
+%ifarch x86_64
+        --slave %{_libdir}/nvidia nvidia-lib64 %{nvidia_libdir} \
+        --slave %{_prefix}/lib/nvidia nvidia-lib %{nvidia_libdir32} \
+%else
+        --slave %{_prefix}/lib/nvidia nvidia-lib %{nvidia_libdir}   \
+%endif
 	--slave %{_libdir}/vdpau/libvdpau_nvidia.so.1 %{_lib}vdpau_nvidia.so.1 %{nvidia_libdir}/vdpau/libvdpau_nvidia.so.%{version} \
 	--slave %{_sysconfdir}/OpenCL/vendors/nvidia.icd nvidia.icd %{_sysconfdir}/%{drivername}/nvidia.icd \
 %ifarch %{biarches}
@@ -1078,7 +1084,7 @@ modprobe -r nvidia > /dev/null 2>&1 || t
 %dir %{nvidia_libdir}
 %dir %{nvidia_libdir}/tls
 %dir %{nvidia_libdir}/vdpau
-%{_libdir}/nvidia
+#%{_libdir}/nvidia
 %if ! %{with external_glvnd}
 %{nvidia_libdir}/libGL.so.1.0.0
 %{nvidia_libdir}/libGL.so.1
@@ -1122,7 +1128,7 @@ modprobe -r nvidia > /dev/null 2>&1 || t
 %dir %{nvidia_libdir32}
 %dir %{nvidia_libdir32}/tls
 %dir %{nvidia_libdir32}/vdpau
-%{_prefix}/lib/nvidia
+#%{_prefix}/lib/nvidia
 %if ! %{with external_glvnd}
 %{nvidia_libdir32}/libGL.so.1.0.0
 %{nvidia_libdir32}/libGL.so.1
