--- a/nvidia390.spec	2022-06-10 08:57:37.000000000 +0300
+++ b/nvidia390.spec	2022-06-25 13:49:53.312579892 +0300
@@ -168,7 +168,6 @@ Requires:	auto-krokodil-filesystem
 %if %{with binary_kernel_modules}
 %kroko_req_modules_in_nvidia -k 5.10 -f generic -j %{version_j} -n %{version_n}
 %endif
-Requires:	%{drivername}-cuda-opencl = %{EVRD}
 Requires:	(kernel-modules-%{name}-5.10.118-1-generic if (kernel-5.10-generic = 5.10.118-1))
 Requires:	(kernel-modules-%{name}-5.15.43-1-generic if (kernel-5.15-generic = 5.15.43-1))
 Requires:	(kernel-modules-%{name}-5.16.19-1-generic if (kernel-5.16-generic = 5.16.19-1))
@@ -203,10 +202,6 @@ Recommends:	libglvnd-gles
 Recommends:	vulkan
 Requires:	libglvnd
 Provides:	nvidia = %{EVRD}
-Conflicts:	nvidia-current
-Conflicts:	nvidia470-core
-Conflicts:	nvidia510-core
-Conflicts:	nvidia515-core
 Recommends:	nvidia-additional-settings
 
 %description -n %{driverpkgname}
@@ -818,10 +813,10 @@ done
 
 # Create the convenience symlinks. As the packages with nvidia are mutually
 # exclusive now, there is no need to use alternatives for this.
-ln -s %{nvidia_libdir} %{buildroot}%{_libdir}/nvidia
-%ifarch %{biarches}
-	ln -s %{nvidia_libdir32} %{buildroot}%{_prefix}/lib/nvidia
-%endif
+#ln -s %{nvidia_libdir} %{buildroot}%{_libdir}/nvidia
+#%ifarch %{biarches}
+#	ln -s %{nvidia_libdir32} %{buildroot}%{_prefix}/lib/nvidia
+#%endif
 
 # confirm SONAME; if something else than libvdpau_nvidia.so or libvdpau_nvidia.so.1, adapt .spec as needed:
 [ "$(objdump -p %{buildroot}%{nvidia_libdir}/vdpau/libvdpau_nvidia.so.%{version} | grep SONAME | gawk '{ print $2 }')" = "libvdpau_nvidia.so.1" ]
@@ -1103,7 +1098,6 @@ modprobe -r nvidia > /dev/null 2>&1 || t
 %dir %{nvidia_libdir}
 %dir %{nvidia_libdir}/tls
 %dir %{nvidia_libdir}/vdpau
-%{_libdir}/nvidia
 %if ! %{with external_glvnd}
 %{nvidia_libdir}/libGL.so.1.0.0
 %{nvidia_libdir}/libGL.so.1
@@ -1147,7 +1141,6 @@ modprobe -r nvidia > /dev/null 2>&1 || t
 %dir %{nvidia_libdir32}
 %dir %{nvidia_libdir32}/tls
 %dir %{nvidia_libdir32}/vdpau
-%{_prefix}/lib/nvidia
 %if ! %{with external_glvnd}
 %{nvidia_libdir32}/libGL.so.1.0.0
 %{nvidia_libdir32}/libGL.so.1
