diff -auprN a/nvidia525.spec b/nvidia525.spec
--- a/nvidia525.spec	2023-03-16 23:04:16.000000000 +0300
+++ b/nvidia525.spec	2023-04-14 20:10:30.540693774 +0300
@@ -31,7 +31,7 @@
 %define pkgname %{pkgname64}
 
 %bcond_without force_32bit_deps
-%bcond_without binary_kernel_modules
+#%bcond_without binary_kernel_modules
 
 # no need to provide libraries in %%nvidia_libdir
 %global __provides_exclude ^lib.*\\.so
@@ -47,9 +47,7 @@
 
 %if %{with binary_kernel_modules}
 # e.g.: 5.10.71-generic-1rosa2021.1-x86_64
-%define kabi_510 %(if rpm -q kernel-5.10-generic-devel 1>/dev/null 2>/dev/null; then rpm -ql kernel-5.10-generic-devel | grep -E '^/lib/modules/.*/source$' | tail -n1 | awk -F '/' '{print $4}'; else echo 0; fi)
-%define kabi_515 %(if rpm -q kernel-5.15-generic-devel 1>/dev/null 2>/dev/null; then rpm -ql kernel-5.15-generic-devel | grep -E '^/lib/modules/.*/source$' | tail -n1 | awk -F '/' '{print $4}'; else echo 0; fi)
-%define kabi_61 %(if rpm -q kernel-6.1-generic-devel 1>/dev/null 2>/dev/null; then rpm -ql kernel-6.1-generic-devel | grep -E '^/lib/modules/.*/source$' | tail -n1 | awk -F '/' '{print $4}'; else echo 0; fi)
+%define kabi_61 %(if rpm -q kernel-6.1-desktop-devel 1>/dev/null 2>/dev/null; then rpm -ql kernel-6.1-desktop-devel | grep -E '^/lib/modules/.*/source$' | tail -n1 | awk -F '/' '{print $4}'; else echo 0; fi)
 %define version_j %(echo %{VERSION} | awk -F '.' '{print $1}')
 %define version_n %(echo %{VERSION} | awk -F '.' '{print $2}')
 # For SRPM stage when auto-krokodil-rpm-macros is not installed
@@ -73,8 +71,6 @@
 # n - name
 %define nv_cross_conflict(n:) %{expand:
 Provides:	nvidia-%{-n*} = %{major}
-Conflicts:	nvidia-%{-n*} > %{major}
-Conflicts:	nvidia-%{-n*} < %{major}
 }
 
 Summary:	NVIDIA proprietary X.org driver and libraries
@@ -115,9 +111,7 @@ ExclusiveArch:	%{x86_64}
 BuildRequires:	auto-krokodil-rpm-generators
 %if %{with binary_kernel_modules}
 BuildRequires:	auto-krokodil-rpm-macros
-BuildRequires:	kernel-5.10-generic-devel
-BuildRequires:	kernel-5.15-generic-devel
-BuildRequires:	kernel-6.1-generic-devel
+BuildRequires:	kernel-6.1-desktop-devel
 BuildRequires:	findutils
 BuildRequires:	zstd
 %endif
@@ -131,6 +125,7 @@ BuildRequires:	pkgconfig(xxf86vm)
 BuildRequires:	pkgconfig(vdpau) >= 1.1
 BuildRequires:	pkgconfig(libtirpc)
 BuildRequires:	pkgconfig(dbus-1)
+BuildRequires:	lib64pciaccess-devel
 
 %description
 The proprietary driver for NVidia GPUs, long-lived branch.
@@ -153,16 +148,8 @@ Conflicts:	x11-driver-video-%{drivername
 Requires:	%{_lib}vdpau1 >= 1.1
 Requires:	auto-krokodil-filesystem
 %if %{with binary_kernel_modules}
-%kroko_req_modules_in_nvidia -k 5.10 -f generic -j %{version_j} -n %{version_n}
-%kroko_req_modules_in_nvidia -k 5.15 -f generic -j %{version_j} -n %{version_n}
-%kroko_req_modules_in_nvidia -k 6.1 -f generic -j %{version_j} -n %{version_n}
+%kroko_req_modules_in_nvidia -k 6.1 -f desktop -j %{version_j} -n %{version_n}
 %endif
-Requires:	%{drivername}-cuda-opencl = %{EVRD}
-# EOLed kernels built without auto-krokodil
-Requires:	(kernel-modules-%{name}-5.16.19-1-generic if (kernel-5.16-generic = 5.16.19-1))
-Requires:	(kernel-modules-%{name}-5.17.11-1-generic if (kernel-5.17-generic = 5.17.11-1))
-Requires:	(kernel-modules-%{name}-5.16.19-2-generic if (kernel-5.16-generic = 5.16.19-2))
-Requires:	(kernel-modules-%{name}-5.17.11-2-generic if (kernel-5.17-generic = 5.17.11-2))
 
 # https://forum.rosalinux.ru/viewtopic.php?p=106968#p106968
 # XXX Maybe better use Supplements in plasma5-kwin-wayland-nvidia
@@ -370,19 +357,7 @@ rm -f nvidia-settings-%{version}/src/*/*
 
 %if %{with binary_kernel_modules}
 dir0="$PWD"
-cp -r %{pkgname}/kernel %{pkgname}/kernel-5.10
-cp -r %{pkgname}/kernel %{pkgname}/kernel-5.15
 cp -r %{pkgname}/kernel %{pkgname}/kernel-6.1
-pushd %{pkgname}/kernel-5.10
-%make SYSSRC=/lib/modules/%{kabi_510}/source
-mkdir -p "$dir0"/%{kabi_510}
-mv *.ko "$dir0"/%{kabi_510}
-popd
-pushd %{pkgname}/kernel-5.15
-%make SYSSRC=/lib/modules/%{kabi_515}/source
-mkdir -p "$dir0"/%{kabi_515}
-mv *.ko "$dir0"/%{kabi_515}
-popd
 pushd %{pkgname}/kernel-6.1
 %make SYSSRC=/lib/modules/%{kabi_61}/source MODULE_ROOT="$PWD"
 mkdir -p "$dir0"/%{kabi_61}
@@ -403,7 +378,7 @@ export LDFLAGS="%{?ldflags}"
 %install
 
 %if %{with binary_kernel_modules}
-list="%{kabi_510} %{kabi_515} %{kabi_61}"
+list="%{kabi_61}"
 # copied from kernel
 # extract debuginfo
 # TODO: sync location of *.debug files with kernel
@@ -1401,9 +1376,7 @@ modprobe -r nvidia > /dev/null 2>&1 || t
 %{nvidia_libdir32}/libcuda.so.1
 
 %if %{with binary_kernel_modules}
-%kroko_kmod_pkg -r %{kroko_release} -j %{version_j} -n %{version_n} -f generic -k 5.10 -m %(if rpm -q kernel-5.10-generic 1>/dev/null 2>/dev/null; then rpm -q --qf '%%{version}\n' kernel-5.10-generic | tail -n1 | awk -F '.' '{print $3}'; else echo 0; fi) -p %{kabi_510} -s %{NAME} -c .zst
-%kroko_kmod_pkg -r %{kroko_release} -j %{version_j} -n %{version_n} -f generic -k 5.15 -m %(if rpm -q kernel-5.15-generic 1>/dev/null 2>/dev/null; then rpm -q --qf '%%{version}\n' kernel-5.15-generic | tail -n1 | awk -F '.' '{print $3}'; else echo 0; fi) -p %{kabi_515} -s %{NAME} -c .zst
-%kroko_kmod_pkg -r %{kroko_release} -j %{version_j} -n %{version_n} -f generic -k 6.1 -m %(if rpm -q kernel-6.1-generic 1>/dev/null 2>/dev/null; then rpm -q --qf '%%{version}\n' kernel-6.1-generic | tail -n1 | awk -F '.' '{print $3}'; else echo 0; fi) -p %{kabi_61} -s %{NAME} -c .zst
+%kroko_kmod_pkg -r %{kroko_release} -j %{version_j} -n %{version_n} -f desktop -k 6.1 -m %(if rpm -q kernel-6.1-desktop 1>/dev/null 2>/dev/null; then rpm -q --qf '%%{version}\n' kernel-6.1-desktop | tail -n1 | awk -F '.' '{print $3}'; else echo 0; fi) -p %{kabi_61} -s %{NAME} -c .zst
 
 %files debug
 %{kroko_kmods_dir}/*/*.debug
