diff -aupr a/kernel.spec b/kernel.spec
--- a/kernel.spec	2020-04-05 18:31:46.000000000 +0300
+++ b/kernel.spec	2020-04-12 13:12:13.756099767 +0300
@@ -1,7 +1,7 @@
 %define kernelversion	5
 %define patchlevel	4
 # sublevel is used for stable-based kernels
-%define sublevel	74
+%define sublevel	78
 
 # Release number. Increase this before a rebuild.
 %define rpmrel		1
@@ -86,6 +86,19 @@
 %define build_modxz		1
 # End of user definitions
 
+# MagOS variables
+%define build_magos		1
+%define build_doc               0
+%define build_source            0
+%define build_devel             1
+%define build_debug             0
+%define build_nrj_desktop	1
+%define build_nrj_laptop	0
+%define build_perf              0
+%define build_cpupower          0
+%define build_modxz             0
+%define build_headers           1
+
 # buildtime flags
 %{?_without_doc: %global build_doc 0}
 %{?_without_devel: %global build_devel 0}
@@ -183,6 +196,11 @@ Patch109:	fs-aufs.patch
 # Other patches
 Patch110:	objtool-sync-check.sh-set-the-exit-code-explicitly.patch
 ####################################################################
+# MagOS patches section
+Source880:	magos-logo.patch
+Source881:	magos-config.patch
+
+####################################################################
 
 Autoreqprov: 	no
 
@@ -243,7 +261,7 @@ Release:	%{fakerel}
 
 Provides:	kernel = %{kverrel}
 Provides:	kernel = %{kernelversion}.%{patchlevel}
-Provides:	kernel-%{flavour} = %{kverrel}
+Provides:	kernel-%{flavour}
 Provides:	alsa = 1.0.27
 Provides:	should-restart = system
 
@@ -682,6 +700,10 @@ echo "Creating the kernel configuration
 
 # Configs
 cp %{SOURCE110} .config
+cp %{SOURCE880} .
+patch -p1 -i %{SOURCE880}
+cp %{SOURCE881} .
+patch -p1 -i %{SOURCE881}
 
 # Disable ASLR for 32-bit systems because it does not play well with
 # hibernate.
@@ -936,7 +958,8 @@ cp -a %{temp_root} %{buildroot}
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if !%{build_magos}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
