diff -auprN a/kernel.spec b/kernel.spec
--- a/kernel.spec	2016-09-21 06:46:31.000000000 +0300
+++ b/kernel.spec	2016-09-21 21:16:20.000000000 +0300
@@ -71,6 +71,18 @@
 %define build_modxz		1
 # End of user definitions
 
+# MagOS variables
+%define build_magos		1
+%define build_doc               0
+%define build_source            0
+%define build_devel             1
+%define build_debug             0
+%define build_nrj_desktop	1
+%define build_nrj_laptop	0
+%define build_perf              0
+%define build_cpupower          0
+%define build_modxz             0
+
 # buildtime flags
 %{?_without_nrj_desktop: %global build_nrj_desktop 0}
 %{?_without_nrj_laptop: %global build_nrj_laptop 0}
@@ -227,6 +239,11 @@ Patch118:	0003-block-bfq-add-Early-Queue
 # then, it seems, even if we place "Patch: <..>" under a conditional.
 Source701:	sanitize-memory.patch
 ####################################################################
+# MagOS patches section
+Source880:	magos-logo.patch
+Source881:	magos-config.patch
+
+####################################################################
 
 # Defines for the things that are needed for all the kernels
 #
@@ -630,6 +647,11 @@ cp %{SOURCE112} .
 FLAVOURS="${FLAVOURS} nrj-laptop"
 %endif
 
+cp %{SOURCE880} .
+patch -p1 -i %{SOURCE880}
+cp %{SOURCE881} .
+patch -p1 -i %{SOURCE881}
+
 CONFIG_DIR=arch/x86/configs
 mkdir -p "${CONFIG_DIR}"
 
@@ -1123,7 +1145,8 @@ find %{target_source}/include -name .ins
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if !%{build_magos}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
