diff -aupr a/kernel.spec b/kernel.spec
--- a/kernel.spec	2018-11-16 20:50:12.423008197 +0300
+++ b/kernel.spec	2018-11-16 20:52:16.879014148 +0300
@@ -1,7 +1,7 @@
 %define kernelversion	4
 %define patchlevel	9
 # sublevel is now used for -stable patches
-%define sublevel	155
+%define sublevel	197
 
 # Release number. Increase this before a rebuild.
 %define rpmrel		1
@@ -74,6 +74,19 @@
 %define build_modxz		1
 # End of user definitions
 
+# MagOS variables
+%define build_magos		1
+%define build_doc               0
+%define build_source            0
+%define build_devel             1
+%define build_debug             0
+%define build_nrj_desktop	1
+%define build_nrj_laptop	0
+%define build_perf              0
+%define build_cpupower          0
+%define build_modxz             0
+%define build_headers           1
+
 # buildtime flags
 %{?_without_nrj_desktop: %global build_nrj_desktop 0}
 %{?_without_nrj_laptop: %global build_nrj_laptop 0}
@@ -237,6 +250,11 @@ Patch303:	random-make-getrandom-ready-ea
 # then, it seems, even if we place "Patch: <..>" under a conditional.
 Source701:	sanitize-memory.patch
 ####################################################################
+# MagOS patches section
+Source880:	magos-logo.patch
+Source881:	magos-config.patch
+
+####################################################################
 
 # Defines for the things that are needed for all the kernels
 #
@@ -639,6 +657,11 @@ cp %{SOURCE112} .
 FLAVOURS="${FLAVOURS} nrj-laptop"
 %endif
 
+cp %{SOURCE880} .
+patch -p1 -i %{SOURCE880}
+cp %{SOURCE881} .
+patch -p1 -i %{SOURCE881}
+
 CONFIG_DIR=arch/x86/configs
 mkdir -p "${CONFIG_DIR}"
 
@@ -1148,7 +1171,8 @@ find %{target_source}/include -name .ins
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if !%{build_magos}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
