--- kernel-3.6.7-1.spec_orig	2012-11-21 00:21:05.000000000 +0400
+++ kernel-3.6.7-1.spec	2012-11-25 18:50:54.828023364 +0400
@@ -301,6 +301,38 @@ Packager: Nicolo' Costanza <abitrules@ya
 %{?_without_kirkwood: %global build_kirkwood 0}
 %{?_without_versatile: %global build_versatile 0}
 
+%define build_modgz                     0
+%{?_with_modgz: %global build_modgz 1}
+# MagOS Linux settings
+%define build_for_magos 0
+%{?_with_build_for_magos: %global build_for_magos 1}
+%if %build_for_magos
+%define build_desktop                   0
+%define build_netbook                   0
+%define build_server                    0
+%define build_desktop_pae               0
+%define build_netbook_pae               0
+%define build_nrj_desktop               0
+%define build_nrj_realtime              0
+%define build_nrj_laptop                0
+%define build_nrj_netbook               0
+%define build_nrj_realtime_pae          0
+%define build_nrj_laptop_pae            0
+%define build_nrj_netbook_pae           0
+%define build_nrj_netbook_atom          0
+%define build_nrj_netbook_atom_pae      0
+%define build_nrj_desktop_core2         0
+%define build_nrj_desktop_core2_pae     0
+%define build_doc                       0
+%define build_source                    0
+%define build_debug                     0
+%define build_perf                      0
+%define build_cpupower                  0
+%define build_modxz                     0
+%define build_modgz                     0
+%define aufs_version                    3.6-20121119
+%endif
+
 # For the .nosrc.rpm
 %define build_nosrc 	0
 %{?_with_nosrc: %global build_nosrc 1}
@@ -371,30 +403,35 @@ Source100: 	linux-%{patch_ver}.tar.xz
 
 %if %kpatch
 %if %sublevel
-Patch2:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/stable-review/patch-%{kversion}-%{kpatch}.bz2
+Patch2:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/stable-review/patch-%{kversion}-%{kpatch}.xz
 Source11:	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/stable-review/patch-%{kversion}-%{kpatch}.sign
 %else
-Patch1:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/testing/patch-%{kernelversion}.%{patchlevel}-%{kpatch}.bz2
+Patch1:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/testing/patch-%{kernelversion}.%{patchlevel}-%{kpatch}.xz
 Source10: 	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/testing/patch-%{kernelversion}.%{patchlevel}-%{kpatch}.sign
 %endif
 %endif
 %if %kgit
-Patch2:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/snapshots/patch-%{kernelversion}.%{patchlevel}-%{kpatch}-git%{kgit}.bz2
+Patch2:		ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/snapshots/patch-%{kernelversion}.%{patchlevel}-%{kpatch}-git%{kgit}.xz
 Source11: 	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/snapshots/patch-%{kernelversion}.%{patchlevel}-%{kpatch}-git%{kgit}.sign
 %endif
 %if %sublevel
 %if %kpatch
 %define prev_sublevel %(expr %{sublevel} - 1)
 %if %prev_sublevel
-Patch1:   	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kernelversion}.%{patchlevel}.%{prev_sublevel}.bz2
+Patch1:   	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kernelversion}.%{patchlevel}.%{prev_sublevel}.xz
 Source10: 	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kernelversion}.%{patchlevel}.%{prev_sublevel}.sign
 %endif
 %else
-Patch1:   	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kversion}.bz2
+Patch1:   	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kversion}.xz
 Source10: 	ftp://ftp.kernel.org/pub/linux/kernel/v%{kernelversion}.%{patchlevel}/patch-%{kversion}.sign
 %endif
 %endif
 
+%if %build_for_magos
+Patch101:       linux-%{patch_ver}-magos.patch
+Patch102:       patch_wireless-add_back_sysfs_directory.patch
+%endif
+
 #END
 ####################################################################
 
@@ -457,6 +494,8 @@ BuildRequires:	module-init-tools
 
 BuildRequires: 	gcc 
 
+%if %build_for_magos
+%else
 # for perf, cpufreq and other tools
 BuildRequires:		elfutils-devel
 BuildRequires:		zlib-devel
@@ -472,6 +511,7 @@ BuildRequires:		docbook-style-xsl
 BuildRequires:		pkgconfig(gtk+-2.0)
 BuildRequires:		flex
 BuildRequires:		bison
+%endif
 
 %ifarch %{arm}
 BuildRequires:		uboot-mkimage
@@ -1079,6 +1119,16 @@ cd %src_dir
 %if %kgit
 %patch2 -p1
 %endif
+%if %build_for_magos
+%patch101 -p1 -d %{patches_dir}
+%patch102 -p1
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-%{aufs_version}.patch"            %{patches_dir}patchesQL/AU01_aufs3-%{aufs_version}.patch
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-base-%{aufs_version}.patch"       %{patches_dir}patchesQL/AU02_aufs3-base-%{aufs_version}.patch
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-kbuild-%{aufs_version}.patch"     %{patches_dir}patchesQL/AU03_aufs3-kbuild-%{aufs_version}.patch
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-loopback-%{aufs_version}.patch"   %{patches_dir}patchesQL/AU04_aufs3-loopback-%{aufs_version}.patch
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-proc_map-%{aufs_version}.patch"   %{patches_dir}patchesQL/AU05_aufs3-proc_map-%{aufs_version}.patch
+cp -pf "%{patches_dir}queue/new for %{kversion}/AUFS3/aufs3-standalone-%{aufs_version}.patch" %{patches_dir}patchesQL/AU06_aufs3-standalone-%{aufs_version}.patch
+%endif
 
 %{patches_dir}/scripts/apply_patchesQL
 %{patches_dir}/scripts/apply_patches
@@ -1707,7 +1757,8 @@ rm -rf %{target_source}/.tmp_depmod/
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if %{build_modgz}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
