--- a/kernel-3.7.10-1.spec	2013-03-05 23:31:17.000000000 +0400
+++ b/kernel-3.7.10-1.spec	2013-03-22 18:09:00.000000000 +0400
@@ -301,6 +301,42 @@ Packager: Nicolo' Costanza <abitrules@ya
 %{?_without_kirkwood: %global build_kirkwood 0}
 %{?_without_versatile: %global build_versatile 0}
 
+%define build_modgz                     0
+%{?_with_modgz: %global build_modgz 1}
+# MagOS Linux settings
+%define build_for_magos 1
+%{?_with_build_for_magos: %global build_for_magos 1}
+%if %build_for_magos
+%define build_doc                       0
+%define build_source                    0
+%define build_devel             	1
+%define build_debug                     0
+%define build_desktop586                0
+%define build_desktop                   0
+%define build_netbook                   0
+%define build_server                    0
+%define build_desktop_pae               0
+%define build_netbook_pae               0
+%define build_nrj_desktop586            0
+%define build_nrj_desktop               0
+%define build_nrj_realtime              0
+%define build_nrj_laptop                0
+%define build_nrj_netbook               0
+%define build_nrj_desktop_pae           1
+%define build_nrj_realtime_pae          0
+%define build_nrj_laptop_pae            0
+%define build_nrj_netbook_pae           0
+%define build_nrj_netbook_atom          0
+%define build_nrj_netbook_atom_pae      0
+%define build_nrj_desktop_core2         0
+%define build_nrj_desktop_core2_pae     0
+%define build_perf                      0
+%define build_cpupower                  0
+%define build_modxz                     0
+%define build_modgz                     0
+%define aufs_version                    3.6-20121119
+%endif
+
 # For the .nosrc.rpm
 %define build_nosrc 	0
 %{?_with_nosrc: %global build_nosrc 1}
@@ -395,6 +431,11 @@ Source10: 	ftp://ftp.kernel.org/pub/linu
 %endif
 %endif
 
+%if %build_for_magos
+Patch101:       linux-%{patch_ver}-magos.patch
+#Patch102:       patch_wireless-add_back_sysfs_directory.patch
+%endif
+
 #END
 ####################################################################
 
@@ -454,6 +495,8 @@ BuildRequires:	module-init-tools
 
 BuildRequires: 	gcc 
 
+%if %build_for_magos
+%else
 # for perf, cpufreq and other tools
 BuildRequires:		elfutils-devel
 BuildRequires:		zlib-devel
@@ -469,6 +512,7 @@ BuildRequires:		docbook-style-xsl
 BuildRequires:		pkgconfig(gtk+-2.0)
 BuildRequires:		flex
 BuildRequires:		bison
+%endif
 
 %ifarch %{arm}
 BuildRequires:		uboot-mkimage
@@ -1084,6 +1128,9 @@ cd %src_dir
 %if %kgit
 %patch2 -p1
 %endif
+%if %build_for_magos
+%patch101 -p1 -d %{patches_dir}
+%endif
 
 %{patches_dir}/scripts/apply_patches
 %{patches_dir}/scripts/apply_patches-others
@@ -1717,7 +1764,8 @@ rm -rf %{target_source}/.tmp_depmod/
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if %{build_modgz}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
