diff -aupr a/kernel-3.10.61-1.spec b/kernel-3.10.61-1.spec
--- a/kernel-3.10.61-1.spec	2014-08-18 06:54:31.000000000 +0400
+++ b/kernel-3.10.61-1.spec	2014-08-18 18:11:11.298832616 +0400
@@ -320,6 +320,48 @@ Packager: Nicolo' Costanza <abitrules@ya
 %{?_without_kirkwood: %global build_kirkwood 0}
 %{?_without_versatile: %global build_versatile 0}
 
+%define build_modgz                     0
+%{?_with_modgz: %global build_modgz 1}
+# MagOS Linux settings
+%define build_for_magos 1
+%{?_with_build_for_magos: %global build_for_magos 1}
+%if %build_for_magos
+%define build_doc                       0
+%define build_source                    0
+%define build_devel             	1
+%define build_debug                     0
+%define build_desktop586                0
+%define build_desktop                   0
+%define build_netbook                   0
+%define build_server                    0
+%define build_desktop_pae               0
+%define build_netbook_pae               0
+%define build_nrj_desktop586            0
+%define build_nrj_realtime              0
+%define build_nrj_laptop                0
+%define build_nrj_netbook               0
+%ifarch %{ix86}
+%define build_nrj_desktop_pae           1
+%define build_nrj_desktop               0
+%endif
+%ifarch x86_64
+%define build_nrj_desktop_pae           0
+%define build_nrj_desktop               1
+%endif
+%define build_nrj_realtime_pae          0
+%define build_nrj_laptop_pae            0
+%define build_nrj_netbook_pae           0
+%define build_nrj_netbook_atom          0
+%define build_nrj_netbook_atom_pae      0
+%define build_nrj_desktop_core2         0
+%define build_nrj_desktop_core2_pae     0
+%define build_perf                      0
+%define build_cpupower                  0
+%define build_modxz                     0
+%define build_modgz                     0
+%define aufs_version                    3.6-20121119
+%endif
+
 # For the .nosrc.rpm
 %define build_nosrc 	0
 %{?_with_nosrc: %global build_nosrc 1}
@@ -494,6 +536,11 @@ Source10: 	ftp://ftp.kernel.org/pub/linu
 %endif
 %endif
 
+%if %build_for_magos
+Patch101:       linux-%{patch_ver}-magos.patch
+# Source102:      aufs3-3.9-allin1.patch
+%endif
+
 #END
 ####################################################################
 
@@ -629,6 +676,8 @@ BuildRequires:  numa-devel
 BuildRequires:  libunwind-devel
 %endif
 
+%if %build_for_magos
+%else
 # for perf, cpufreq and other tools
 BuildRequires:		elfutils-devel
 BuildRequires:		zlib-devel
@@ -644,6 +693,7 @@ BuildRequires:		docbook-style-xsl
 BuildRequires:		pkgconfig(gtk+-2.0)
 BuildRequires:		flex
 BuildRequires:		bison
+%endif
 
 %ifarch %{arm}
 BuildRequires:		uboot-mkimage
@@ -1306,6 +1356,10 @@ cd %src_dir
 %if %kgit
 %patch2 -p1
 %endif
+%if %build_for_magos
+%patch101 -p1 -d %{patches_dir}
+# cp -pf  %{SOURCE102} %{patches_dir}/patches-extras
+%endif
 
 %{patches_dir}/scripts/apply_patches
 %{patches_dir}/scripts/apply_patches-backlight
@@ -2024,7 +2078,8 @@ rm -rf %{target_source}/.tmp_depmod/
 # compressing modules
 %if %{build_modxz}
 find %{target_modules} -name "*.ko" | %kxargs xz -6e
-%else
+%endif
+%if %{build_modgz}
 find %{target_modules} -name "*.ko" | %kxargs gzip -9
 %endif
 
