diff -auprN a/Documentation/ABI/testing/debugfs-aufs b/Documentation/ABI/testing/debugfs-aufs
diff -auprN a/Documentation/ABI/testing/sysfs-aufs b/Documentation/ABI/testing/sysfs-aufs
diff -auprN a/fs/aufs/aufs.h b/fs/aufs/aufs.h
diff -auprN a/fs/aufs/branch.c b/fs/aufs/branch.c
diff -auprN a/fs/aufs/branch.h b/fs/aufs/branch.h
diff -auprN a/fs/aufs/conf.mk b/fs/aufs/conf.mk
diff -auprN a/fs/aufs/cpup.c b/fs/aufs/cpup.c
diff -auprN a/fs/aufs/cpup.h b/fs/aufs/cpup.h
diff -auprN a/fs/aufs/dbgaufs.c b/fs/aufs/dbgaufs.c
diff -auprN a/fs/aufs/dbgaufs.h b/fs/aufs/dbgaufs.h
diff -auprN a/fs/aufs/dcsub.c b/fs/aufs/dcsub.c
diff -auprN a/fs/aufs/dcsub.h b/fs/aufs/dcsub.h
diff -auprN a/fs/aufs/debug.c b/fs/aufs/debug.c
diff -auprN a/fs/aufs/debug.h b/fs/aufs/debug.h
diff -auprN a/fs/aufs/dentry.c b/fs/aufs/dentry.c
diff -auprN a/fs/aufs/dentry.h b/fs/aufs/dentry.h
diff -auprN a/fs/aufs/dinfo.c b/fs/aufs/dinfo.c
diff -auprN a/fs/aufs/dir.c b/fs/aufs/dir.c
diff -auprN a/fs/aufs/dir.h b/fs/aufs/dir.h
diff -auprN a/fs/aufs/dynop.c b/fs/aufs/dynop.c
diff -auprN a/fs/aufs/dynop.h b/fs/aufs/dynop.h
diff -auprN a/fs/aufs/export.c b/fs/aufs/export.c
diff -auprN a/fs/aufs/file.c b/fs/aufs/file.c
diff -auprN a/fs/aufs/file.h b/fs/aufs/file.h
diff -auprN a/fs/aufs/finfo.c b/fs/aufs/finfo.c
diff -auprN a/fs/aufs/f_op.c b/fs/aufs/f_op.c
diff -auprN a/fs/aufs/f_op_sp.c b/fs/aufs/f_op_sp.c
diff -auprN a/fs/aufs/fstype.h b/fs/aufs/fstype.h
diff -auprN a/fs/aufs/hfsnotify.c b/fs/aufs/hfsnotify.c
diff -auprN a/fs/aufs/hfsplus.c b/fs/aufs/hfsplus.c
diff -auprN a/fs/aufs/hnotify.c b/fs/aufs/hnotify.c
diff -auprN a/fs/aufs/iinfo.c b/fs/aufs/iinfo.c
diff -auprN a/fs/aufs/inode.c b/fs/aufs/inode.c
diff -auprN a/fs/aufs/inode.h b/fs/aufs/inode.h
diff -auprN a/fs/aufs/ioctl.c b/fs/aufs/ioctl.c
diff -auprN a/fs/aufs/i_op_add.c b/fs/aufs/i_op_add.c
diff -auprN a/fs/aufs/i_op.c b/fs/aufs/i_op.c
diff -auprN a/fs/aufs/i_op_del.c b/fs/aufs/i_op_del.c
diff -auprN a/fs/aufs/i_op_ren.c b/fs/aufs/i_op_ren.c
diff -auprN a/fs/aufs/Kconfig b/fs/aufs/Kconfig
diff -auprN a/fs/aufs/loop.c b/fs/aufs/loop.c
diff -auprN a/fs/aufs/loop.h b/fs/aufs/loop.h
diff -auprN a/fs/aufs/magic.mk b/fs/aufs/magic.mk
diff -auprN a/fs/aufs/Makefile b/fs/aufs/Makefile
diff -auprN a/fs/aufs/module.c b/fs/aufs/module.c
diff -auprN a/fs/aufs/module.h b/fs/aufs/module.h
diff -auprN a/fs/aufs/opts.c b/fs/aufs/opts.c
diff -auprN a/fs/aufs/opts.h b/fs/aufs/opts.h
diff -auprN a/fs/aufs/plink.c b/fs/aufs/plink.c
diff -auprN a/fs/aufs/poll.c b/fs/aufs/poll.c
diff -auprN a/fs/aufs/procfs.c b/fs/aufs/procfs.c
diff -auprN a/fs/aufs/rdu.c b/fs/aufs/rdu.c
diff -auprN a/fs/aufs/rwsem.h b/fs/aufs/rwsem.h
diff -auprN a/fs/aufs/sbinfo.c b/fs/aufs/sbinfo.c
diff -auprN a/fs/aufs/spl.h b/fs/aufs/spl.h
diff -auprN a/fs/aufs/super.c b/fs/aufs/super.c
diff -auprN a/fs/aufs/super.h b/fs/aufs/super.h
diff -auprN a/fs/aufs/sysaufs.c b/fs/aufs/sysaufs.c
diff -auprN a/fs/aufs/sysaufs.h b/fs/aufs/sysaufs.h
diff -auprN a/fs/aufs/sysfs.c b/fs/aufs/sysfs.c
diff -auprN a/fs/aufs/sysrq.c b/fs/aufs/sysrq.c
diff -auprN a/fs/aufs/vdir.c b/fs/aufs/vdir.c
diff -auprN a/fs/aufs/vfsub.c b/fs/aufs/vfsub.c
diff -auprN a/fs/aufs/vfsub.h b/fs/aufs/vfsub.h
diff -auprN a/fs/aufs/wbr_policy.c b/fs/aufs/wbr_policy.c
diff -auprN a/fs/aufs/whout.c b/fs/aufs/whout.c
diff -auprN a/fs/aufs/whout.h b/fs/aufs/whout.h
diff -auprN a/fs/aufs/wkq.c b/fs/aufs/wkq.c
diff -auprN a/fs/aufs/wkq.h b/fs/aufs/wkq.h
diff -auprN a/fs/aufs/xino.c b/fs/aufs/xino.c
diff -auprN a/fs/file_table.c b/fs/file_table.c
diff -auprN a/fs/inode.c b/fs/inode.c
diff -auprN a/fs/Kconfig b/fs/Kconfig
diff -auprN a/fs/Makefile b/fs/Makefile
diff -auprN a/fs/namei.c b/fs/namei.c
diff -auprN a/fs/namei.c.orig b/fs/namei.c.orig
diff -auprN a/fs/namespace.c b/fs/namespace.c
diff -auprN a/fs/namespace.c.orig b/fs/namespace.c.orig
diff -auprN a/fs/notify/group.c b/fs/notify/group.c
diff -auprN a/fs/notify/mark.c b/fs/notify/mark.c
diff -auprN a/fs/open.c b/fs/open.c
diff -auprN a/fs/splice.c b/fs/splice.c
diff -auprN a/include/linux/aufs_type.h b/include/linux/aufs_type.h
--- a/include/linux/aufs_type.h	1970-01-01 03:00:00.000000000 +0300
+++ b/include/linux/aufs_type.h	2011-12-13 21:48:00.000000000 +0400
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2005-2011 Junjiro R. Okajima
+ *
+ * This program, aufs is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef __AUFS_TYPE_H__
+#define __AUFS_TYPE_H__
+
+#include <linux/ioctl.h>
+#include <linux/kernel.h>
+#include <linux/limits.h>
+#ifdef __KERNEL__
+#include <linux/types.h>
+#else
+#include <stdint.h>
+#include <sys/types.h>
+#endif
+
+#define AUFS_VERSION	"2.2-standalone.tree-39-20111114"
+
+/* todo? move this to linux-2.6.19/include/magic.h */
+#define AUFS_SUPER_MAGIC	('a' << 24 | 'u' << 16 | 'f' << 8 | 's')
+
+/* ---------------------------------------------------------------------- */
+
+#ifdef CONFIG_AUFS_BRANCH_MAX_127
+typedef int8_t aufs_bindex_t;
+#define AUFS_BRANCH_MAX 127
+#else
+typedef int16_t aufs_bindex_t;
+#ifdef CONFIG_AUFS_BRANCH_MAX_511
+#define AUFS_BRANCH_MAX 511
+#elif defined(CONFIG_AUFS_BRANCH_MAX_1023)
+#define AUFS_BRANCH_MAX 1023
+#elif defined(CONFIG_AUFS_BRANCH_MAX_32767)
+#define AUFS_BRANCH_MAX 32767
+#endif
+#endif
+
+#ifdef __KERNEL__
+#ifndef AUFS_BRANCH_MAX
+#error unknown CONFIG_AUFS_BRANCH_MAX value
+#endif
+#endif /* __KERNEL__ */
+
+/* ---------------------------------------------------------------------- */
+
+#define AUFS_NAME		"aufs"
+#define AUFS_FSTYPE		AUFS_NAME
+
+#define AUFS_ROOT_INO		2
+#define AUFS_FIRST_INO		11
+
+#define AUFS_WH_PFX		".wh."
+#define AUFS_WH_PFX_LEN		((int)sizeof(AUFS_WH_PFX) - 1)
+#define AUFS_WH_TMP_LEN		4
+/* a limit for rmdir/rename a dir */
+#define AUFS_MAX_NAMELEN	(NAME_MAX \
+				- AUFS_WH_PFX_LEN * 2	/* doubly whiteouted */\
+				- 1			/* dot */\
+				- AUFS_WH_TMP_LEN)	/* hex */
+#define AUFS_XINO_FNAME		"." AUFS_NAME ".xino"
+#define AUFS_XINO_DEFPATH	"/tmp/" AUFS_XINO_FNAME
+#define AUFS_XINO_TRUNC_INIT	64 /* blocks */
+#define AUFS_XINO_TRUNC_STEP	4  /* blocks */
+#define AUFS_DIRWH_DEF		3
+#define AUFS_RDCACHE_DEF	10 /* seconds */
+#define AUFS_RDCACHE_MAX	3600 /* seconds */
+#define AUFS_RDBLK_DEF		512 /* bytes */
+#define AUFS_RDHASH_DEF		32
+#define AUFS_WKQ_NAME		AUFS_NAME "d"
+#define AUFS_MFS_DEF_SEC	30 /* seconds */
+#define AUFS_MFS_MAX_SEC	3600 /* seconds */
+#define AUFS_PLINK_WARN		100 /* number of plinks */
+
+/* pseudo-link maintenace under /proc */
+#define AUFS_PLINK_MAINT_NAME	"plink_maint"
+#define AUFS_PLINK_MAINT_DIR	"fs/" AUFS_NAME
+#define AUFS_PLINK_MAINT_PATH	AUFS_PLINK_MAINT_DIR "/" AUFS_PLINK_MAINT_NAME
+
+#define AUFS_DIROPQ_NAME	AUFS_WH_PFX ".opq" /* whiteouted doubly */
+#define AUFS_WH_DIROPQ		AUFS_WH_PFX AUFS_DIROPQ_NAME
+
+#define AUFS_BASE_NAME		AUFS_WH_PFX AUFS_NAME
+#define AUFS_PLINKDIR_NAME	AUFS_WH_PFX "plnk"
+#define AUFS_ORPHDIR_NAME	AUFS_WH_PFX "orph"
+
+/* doubly whiteouted */
+#define AUFS_WH_BASE		AUFS_WH_PFX AUFS_BASE_NAME
+#define AUFS_WH_PLINKDIR	AUFS_WH_PFX AUFS_PLINKDIR_NAME
+#define AUFS_WH_ORPHDIR		AUFS_WH_PFX AUFS_ORPHDIR_NAME
+
+/* branch permissions and attributes */
+#define AUFS_BRPERM_RW		"rw"
+#define AUFS_BRPERM_RO		"ro"
+#define AUFS_BRPERM_RR		"rr"
+#define AUFS_BRRATTR_WH		"wh"
+#define AUFS_BRWATTR_NLWH	"nolwh"
+
+/* ---------------------------------------------------------------------- */
+
+/* ioctl */
+enum {
+	/* readdir in userspace */
+	AuCtl_RDU,
+	AuCtl_RDU_INO,
+
+	/* pathconf wrapper */
+	AuCtl_WBR_FD,
+
+	/* busy inode */
+	AuCtl_IBUSY
+};
+
+/* borrowed from linux/include/linux/kernel.h */
+#ifndef ALIGN
+#define ALIGN(x, a)		__ALIGN_MASK(x, (typeof(x))(a)-1)
+#define __ALIGN_MASK(x, mask)	(((x)+(mask))&~(mask))
+#endif
+
+/* borrowed from linux/include/linux/compiler-gcc3.h */
+#ifndef __aligned
+#define __aligned(x)			__attribute__((aligned(x)))
+#endif
+
+#ifdef __KERNEL__
+#ifndef __packed
+#define __packed			__attribute__((packed))
+#endif
+#endif
+
+struct au_rdu_cookie {
+	uint64_t	h_pos;
+	int16_t		bindex;
+	uint8_t		flags;
+	uint8_t		pad;
+	uint32_t	generation;
+} __aligned(8);
+
+struct au_rdu_ent {
+	uint64_t	ino;
+	int16_t		bindex;
+	uint8_t		type;
+	uint8_t		nlen;
+	uint8_t		wh;
+	char		name[0];
+} __aligned(8);
+
+static inline int au_rdu_len(int nlen)
+{
+	/* include the terminating NULL */
+	return ALIGN(sizeof(struct au_rdu_ent) + nlen + 1,
+		     sizeof(uint64_t));
+}
+
+union au_rdu_ent_ul {
+	struct au_rdu_ent __user	*e;
+	uint64_t			ul;
+};
+
+enum {
+	AufsCtlRduV_SZ,
+	AufsCtlRduV_End
+};
+
+struct aufs_rdu {
+	/* input */
+	union {
+		uint64_t	sz;	/* AuCtl_RDU */
+		uint64_t	nent;	/* AuCtl_RDU_INO */
+	};
+	union au_rdu_ent_ul	ent;
+	uint16_t		verify[AufsCtlRduV_End];
+
+	/* input/output */
+	uint32_t		blk;
+
+	/* output */
+	union au_rdu_ent_ul	tail;
+	/* number of entries which were added in a single call */
+	uint64_t		rent;
+	uint8_t			full;
+	uint8_t			shwh;
+
+	struct au_rdu_cookie	cookie;
+} __aligned(8);
+
+/* ---------------------------------------------------------------------- */
+
+struct aufs_wbr_fd {
+	uint32_t	oflags;
+	int16_t		brid;
+} __aligned(8);
+
+/* ---------------------------------------------------------------------- */
+
+struct aufs_ibusy {
+	uint64_t	ino, h_ino;
+	int16_t		bindex;
+} __aligned(8);
+
+/* ---------------------------------------------------------------------- */
+
+#define AuCtlType		'A'
+#define AUFS_CTL_RDU		_IOWR(AuCtlType, AuCtl_RDU, struct aufs_rdu)
+#define AUFS_CTL_RDU_INO	_IOWR(AuCtlType, AuCtl_RDU_INO, struct aufs_rdu)
+#define AUFS_CTL_WBR_FD		_IOW(AuCtlType, AuCtl_WBR_FD, \
+				     struct aufs_wbr_fd)
+#define AUFS_CTL_IBUSY		_IOWR(AuCtlType, AuCtl_IBUSY, struct aufs_ibusy)
+
+#endif /* __AUFS_TYPE_H__ */
diff -auprN a/include/linux/Kbuild b/include/linux/Kbuild
--- a/include/linux/Kbuild	2011-10-28 14:15:11.000000000 +0400
+++ b/include/linux/Kbuild	2011-12-13 21:47:06.000000000 +0400
@@ -64,6 +64,7 @@ header-y += atmppp.h
 header-y += atmsap.h
 header-y += atmsvc.h
 header-y += audit.h
+header-y += aufs_type.h
 header-y += auto_fs.h
 header-y += auto_fs4.h
 header-y += auxvec.h
diff -auprN a/include/linux/namei.h b/include/linux/namei.h
--- a/include/linux/namei.h	2011-10-28 14:15:11.000000000 +0400
+++ b/include/linux/namei.h	2011-12-13 21:46:42.000000000 +0400
@@ -82,6 +82,7 @@ extern int vfs_path_lookup(struct dentry
 extern struct file *lookup_instantiate_filp(struct nameidata *nd, struct dentry *dentry,
 		int (*open)(struct inode *, struct file *));
 
+extern struct dentry *lookup_hash(struct nameidata *nd);
 extern struct dentry *lookup_one_len(const char *, struct dentry *, int);
 
 extern int follow_down_one(struct path *);
diff -auprN a/include/linux/splice.h b/include/linux/splice.h
--- a/include/linux/splice.h	2011-10-28 14:15:11.000000000 +0400
+++ b/include/linux/splice.h	2011-12-13 21:46:42.000000000 +0400
@@ -89,4 +89,10 @@ extern int splice_grow_spd(struct pipe_i
 extern void splice_shrink_spd(struct pipe_inode_info *,
 				struct splice_pipe_desc *);
 
+extern long do_splice_from(struct pipe_inode_info *pipe, struct file *out,
+			   loff_t *ppos, size_t len, unsigned int flags);
+extern long do_splice_to(struct file *in, loff_t *ppos,
+			 struct pipe_inode_info *pipe, size_t len,
+			 unsigned int flags);
+
 #endif
diff -auprN a/security/commoncap.c b/security/commoncap.c
diff -auprN a/security/device_cgroup.c b/security/device_cgroup.c
diff -auprN a/security/security.c b/security/security.c
