--- a/kernel-desktop.spec	2011-12-13 20:13:17.000000000 +0400
+++ b/kernel-desktop.spec	2011-12-13 22:22:55.000000000 +0400
@@ -5,13 +5,17 @@ Name:          kernel-desktop
 Summary:       The Linux Kernel for Mandriva systems
 License:       GPLv2
 Version:       2.6.39.4
-Release:       %mkrel 4.2
+Release:       %mkrel 4.2flash
 URL:           http://www.kernel.org
 Source:        kernel-desktop-2.6.39.4-4.tar.bz2
 ExclusiveArch: %ix86 x86_64 
 BuildRoot:     %{_tmppath}/%{name}-%{version}-root
+Patch1:        logo_magos.patch
+Patch2:        aufs2-linux.patch
+Patch3:        config.patch
+Patch4:        devel_rpm.patch
 
-%global __debug_package 1
+%global __debug_package 0
 %define debug_package %{nil}
 %define __check_files %{nil}
 
@@ -55,6 +59,10 @@ by some binary object tools like kgdb, p
 
 %prep
 %setup -q -n kernel-desktop-2.6.39.4-4
+%patch1 -p1
+%patch2 -p1
+%patch3 -p1
+%patch4 -p1
 
 %build
 make defconfig
