diff -auprN 2.6.33.5-mnb2/scripts/create_configs 2.6.33.5-mnb2_pae/scripts/create_configs
--- 2.6.33.5-mnb2/scripts/create_configs	2010-06-17 22:47:25.000000000 +0400
+++ 2.6.33.5-mnb2_pae/scripts/create_configs	2010-07-14 11:52:59.000000000 +0400
@@ -26,8 +26,8 @@ my $configs_dir="";
 my @archs = qw(i386 powerpc sparc x86_64);
 
 my @configs = (
-     [ qw(i586 smp 4GB  desktop586) ],
-     [ qw(i686 smp 4GB  desktop) ],
+     [ qw(i586 smp 64GB  desktop586) ],
+     [ qw(i686 smp 64GB  desktop) ],
      [ qw(i686 smp 64GB server) ],
 
      [ qw(powerpc smp normal desktop) ],
