diff -auprN a/etc/dkms/template-dkms-mkrpm.spec b/etc/dkms/template-dkms-mkrpm.spec
--- a/etc/dkms/template-dkms-mkrpm.spec	2012-10-18 06:15:27.000000000 +0400
+++ b/etc/dkms/template-dkms-mkrpm.spec	2013-04-21 08:45:12.000000000 +0400
@@ -14,7 +14,7 @@
 %define _dkmsbinarydir %_localstatedir/lib/dkms-binary
 %define moduledkmsbinarydir %{_dkmsbinarydir}/%{module_name}/%{fake_version}
 
-%define kernel_flavour_rx /(.*)-(\\D.*)/
+%define kernel_flavour_rx /([\\d\\.]*)-(.*)-(\\d.*)/
 %define kernel_flavour_cmd perl -e '$ARGV[0] =~ %kernel_flavour_rx and print "$2\\n"'
 # find latest kernel for each flavour of the defined kernels
 %define kernels_latest %(perl -e 'foreach my $ver (sort(qw(%kernel_versions))) { $ver =~ %kernel_flavour_rx and $kernels{$1} = $ver }; print join(" ", values(%%kernels)) . "\\n"')
