diff -auprN a/systemd.spec b/systemd.spec
--- a/systemd.spec	2013-06-22 17:11:39.000000000 +0400
+++ b/systemd.spec	2013-06-22 17:24:35.000000000 +0400
@@ -99,6 +99,7 @@ Patch507: 0507-reinstate-TIMEOUT-handlin
 Patch508: 0508-udev-Allow-the-udevadm-settle-timeout-to-be-set-via-.patch
 Patch509: 0508-Mageia-Relax-perms-on-sys-kernel-debug-for-lspcidrak.patch
 Patch510: 0509-udev-rules-Apply-SuSE-patch-to-restore-cdrom-cdrw-dv.patch
+Patch600: 0600-systemd-magos.patch
 
 BuildRequires:	autoconf
 BuildRequires:	automake
