--- a/lxde-ctrl-center.spec	2013-04-20 08:49:51.000000000 +0400
+++ b/lxde-ctrl-center.spec	2013-04-20 08:58:12.000000000 +0400
@@ -8,6 +8,7 @@ URL: http://code.google.com/p/lxde-ctrl-
 Source0: http://code.google.com/p/lxde-ctrl-center/files/%{name}-%{version}.tar.gz
 Source1: clouds.png
 Source2: bg.png
+Patch0: magos-po-0.0.6.diff
 BuildRequires: gettext
 Requires: python, pygtk2.0, python-webkitgtk, python-simplejson
 Requires: beesu
@@ -75,6 +76,7 @@ for DrakX and LXDE configuration program
 cp -f %SOURCE1 ./share/%{name}/frontend/images/
 cp -f %SOURCE2 ./share/%{name}/frontend/images/
 sed -i -e 's/"beesu drakconnect"/"nm-connection-editor"/g' ./share/%{name}/items/network
+%apply_patches
 
 %build
 ./make build_pkg
