diff -auprN a/kernel.spec b/kernel.spec
--- a/kernel.spec	2010-03-29 15:24:15.000000000 +0400
+++ b/kernel.spec	2010-03-29 15:23:53.000000000 +0400
@@ -12,7 +12,7 @@
 %define sublevel	31
 
 # Package release
-%define mnbrel          1flash_nopae_
+%define mnbrel          1flash_
 %define realmnbrel      1
 
 # kernel Makefile extraversion is substituted by 
@@ -240,6 +240,7 @@ Source10: 	ftp://ftp.kernel.org/pub/linu
 Patch3:         aufs-linux-flash.patch
 Patch4:         aufs-mnb-flash.patch
 Patch5:         logo_mdk_clut224.patch
+Patch6:         aufs-mnb-pae-flash.patch
 
 #END
 ####################################################################
@@ -655,6 +656,7 @@ cd %src_dir
 rm -f drivers/video/logo/logo_mdk_clut224.ppm
 cd %{patches_dir}
 %patch4 -p1
+%patch6 -p1
 cd %src_dir
 %patch5 -p1
 cd %src_dir
